From: Chad Dupuis <cdupuis@redhat.com>
Date: Fri, 10 Dec 2010 18:55:03 -0500
Subject: [scsi] qla4xxx: mask bits in F/W Options during init
Message-id: <20101210185503.7449.27174.sendpatchset@localhost.localdomain>
Patchwork-id: 30078
O-Subject: [RHEL 5.6 PATCH 1/4] qla4xxx: Masking required bits in Additional F/W
	Options during Initilization.
Bugzilla: 656999

Bugzilla
--------

Bug 656999 (https://bugzilla.redhat.com/show_bug.cgi?id=656999)

Upstream Status
---------------

Will be pushed upstream in our next patch submission.

Description
-----------

>From 0d090f18a729d40d8ded494ee5000f644a4a501a Mon Sep 17 00:00:00 2001
From: Prasanna Mumbai <prasanna.mumbai@qlogic.com>
Date: Thu, 9 Dec 2010 15:06:03 -0800
Subject: [PATCH 1/4] qla4xxx: Masking required bits in Additional F/W Options during Initilization.

Signed-off-by: Prasanna Mumbai <prasanna.mumbai@qlogic.com>

diff --git a/drivers/scsi/qla4xxx/ql4_fw.h b/drivers/scsi/qla4xxx/ql4_fw.h
index 6a93198..c54038a 100644
--- a/drivers/scsi/qla4xxx/ql4_fw.h
+++ b/drivers/scsi/qla4xxx/ql4_fw.h
@@ -473,6 +473,7 @@ struct addr_ctrl_blk {
 	uint8_t res0;	/* 07 */
 	uint16_t eth_mtu_size;	/* 08-09 */
 	uint16_t add_fw_options;	/* 0A-0B */
+#define SERIALIZE_TASK_MGMT		0x0400
 
 	uint8_t hb_interval;	/* 0C */
 	uint8_t inst_num; /* 0D */
diff --git a/drivers/scsi/qla4xxx/ql4_mbx.c b/drivers/scsi/qla4xxx/ql4_mbx.c
index 96a57d2..d3b99c7 100644
--- a/drivers/scsi/qla4xxx/ql4_mbx.c
+++ b/drivers/scsi/qla4xxx/ql4_mbx.c
@@ -534,6 +534,8 @@ int qla4xxx_initialize_fw_cb(struct scsi_qla_host * ha)
 			__constant_cpu_to_le16(FWOPT_ENABLE_CRBDB);
 
 	init_fw_cb->fw_options &= __constant_cpu_to_le16(~FWOPT_TARGET_MODE);
+	init_fw_cb->add_fw_options &=
+				__constant_cpu_to_le16(SERIALIZE_TASK_MGMT);
 
 	if (qla4xxx_set_ifcb(ha, &mbox_cmd[0], &mbox_sts[0], init_fw_cb_dma)
 		!= QLA_SUCCESS) {
